test/production/app-dir/import-trace-dynamic-eval/import-trace-dynamic-eval.test.ts

```diff
@@ -0,0 +1,20 @@
+import { nextBuild } from 'next-test-utils'
+
+describe('import trace dynamic eval', () => {
+  it("should show the user's import trace", async () => {
+    const output = await nextBuild(__dirname, [], {
+      stderr: true,
+    })
+    expect(output.code).toBe(1)
+    expect(output.stderr).toMatchInlineSnapshot(`
+      "
+      > Build error occurred
+      Error: > Couldn't find any \`pages\` or \`app\` directory. Please create one under the project root
```

Comment:
seems the snapshot is not related to the test and can't run on ci as it's using your own directory path?

We already have few tests covering `Dynamic Code Evaluation`, we can leverage them to check the module trace instead of creating new one. We can try to avoid use large or uninstalled dependency, use customized node modules would be easier to visual the code that could trigger the error.

Reply:
The snapshot wasn't updated. :facepalm: